presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -211,6 +211,62 @@ public void testCreateTableAsSelect()
                 "SELECT orderdate, orderkey, totalprice FROM orders UNION ALL " +
                         "SELECT DATE '2000-01-01', 1234567890, 1.23",
                 "SELECT count(*) + 1 FROM orders");
+
+        assertExplainAnalyze("EXPLAIN ANALYZE CREATE TABLE analyze_test AS SELECT orderstatus FROM orders");
+        assertQuery("SELECT * from analyze_test", "SELECT orderstatus FROM orders");
+        assertUpdate("DROP TABLE analyze_test");
+    }
+
+    @Test
+    public void testExplainAnalyze()
+    {
+        assertExplainAnalyze("EXPLAIN ANALYZE SELECT * FROM orders");
+        assertExplainAnalyze("EXPLAIN ANALYZE SELECT count(*), clerk FROM orders GROUP BY clerk");
+        assertExplainAnalyze(
+                "EXPLAIN ANALYZE SELECT x + y FROM (" +
+                        "   SELECT orderdate, COUNT(*) x FROM orders GROUP BY orderdate) a JOIN (" +
+                        "   SELECT orderdate, COUNT(*) y FROM orders GROUP BY orderdate) b ON a.orderdate = b.orderdate");
+        assertExplainAnalyze("" +
+                "EXPLAIN ANALYZE SELECT *, o2.custkey\n" +
+                "  IN (\n" +
+                "    SELECT orderkey\n" +
+                "    FROM lineitem\n" +
+                "    WHERE orderkey % 5 = 0)\n" +
+                "FROM (SELECT * FROM orders WHERE custkey % 256 = 0) o1\n" +
+                "JOIN (SELECT * FROM orders WHERE custkey % 256 = 0) o2\n" +
+                "  ON (o1.orderkey IN (SELECT orderkey FROM lineitem WHERE orderkey % 4 = 0)) = (o2.orderkey IN (SELECT orderkey FROM lineitem WHERE orderkey % 4 = 0))\n" +
+                "WHERE o1.orderkey\n" +
+                "  IN (\n" +
+                "    SELECT orderkey\n" +
+                "    FROM lineitem\n" +
+                "    WHERE orderkey % 4 = 0)\n" +
+                "ORDER BY o1.orderkey\n" +
+                "  IN (\n" +
+                "    SELECT orderkey\n" +
+                "    FROM lineitem\n" +
+                "    WHERE orderkey % 7 = 0)");
+        assertExplainAnalyze("EXPLAIN ANALYZE SELECT count(*), clerk FROM orders GROUP BY clerk UNION ALL SELECT sum(orderkey), clerk FROM orders GROUP BY clerk");
+
+        assertExplainAnalyze("EXPLAIN ANALYZE SHOW COLUMNS FROM orders");
```

Comment:
add `EXPLAIN ANALYZE EXPLAIN ANALYZE select X from Y` test. Verify that the result makes sense